### PR TITLE
Adash event architecture

### PR DESCRIPTION
**Pull request by AbhijeetDash**
With some feedback from other developers, this is a revamp of the current project structure. It makes sure things are aligned to build on top. 
1. Removed Core -> Added Services. 
2. Separated Responsibility of Helper classes, Formater classes and Middleware classes.

Fix: In some cases with groq we were not able to find the replacement code in the file. This fix addresses those issues. 

Add: Added a version flag to the oi command. Now we can run `oi -V | --version` to see the current version.

### CODE REVIEW

**Comment by AbhijeetDash** [comment]
@Zoheb-Malik This is a major refactor of the codebase. It introduces more abstract classes which would allow us to write tests in coming months. We can keep adding abstraction to remaining classes if you approve this refactor.

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
   },
   "scripts": {
     "pnpm": "pnpm",
```
Comment (Zoheb-Malik) [inline package.json:12]: pnpm would be installed globally as per the README.md, so there wouldn't be a need to have a separate script named pnpm

File: package.json [hunk 2]
```diff
@@ -34,6 +35,7 @@
     "globals": "^15.10.0",
     "pnpm": "^9.12.2",
```
Comment (Zoheb-Malik) [inline package.json]: pnpm not needed as a dev dependency as far as I'm aware

File: src/commands/command.start.ts [hunk 3]
```diff
@@ -43,12 +43,68 @@ class Start extends OiCommand {
...
     }
   }
+
+  onAdd(filePath: string, verbose: boolean | undefined): void {
+    if (verbose) {
+      console.log(`File ${filePath} has been added`);
+    }
+  }
+
+  async onFileChanged(filePath: string, verbose: boolean | undefined): Promise<void> {
```
Comment (Zoheb-Malik) [inline src/commands/command.start.ts:84]: Excellent addition here!

File: src/index.ts [hunk 4]
```diff
@@ -8,6 +8,8 @@ import Start from './commands/command.start';
 
 const program = new Command();
 
+program.version('0.0.5');
```
Comment (Zoheb-Malik) [inline src/index.ts:11]: Please explain purpose of this?

File: src/utilis/util.service.dev.ts
Comment (Zoheb-Malik) [inline src/utilis/util.service.dev.ts:1]: The help.code.ts file has a conflict which needs resolving

**Review by Zoheb-Malik — CHANGES_REQUESTED** [review]
I have left a couple of comments, some things to consider, and a conflict that needs addressing. Overall, a great first step in refactoring oi!

File: src/index.ts (hunk 4 above)
Comment (AbhijeetDash) [reply src/index.ts:11]: This allows user to check the version of oi installed on the system.. oi -V or oi --version

**Review by AbhijeetDash — COMMENTED** [review]
Made Required Changes

**Pull request merged**